system/ee/ExpressionEngine/Library/Advisor/TemplateAdvisor.php

```diff
@@ -70,6 +70,12 @@ public function getAllTags()
                 ];
             }
 
+            // remove any EE comments in template first (to prevent tags within comments being flagged as problems)
+            // this can be destructive as we don't make any persistent changes to templates and getAllTags does not
+            // return line numbers of found tags
+            $regex_comments = '/(\{!--\s(?:.|\R)*?\s--\})/';
```

Comment:
In `remove_ee_comments` function, EE is currently using simplier regex:
> "/\{!--.*?--\}/s"
Just wondering, why yours is more complex?